internal/server/handler_v1.go

```diff
@@ -532,20 +532,23 @@ func (s *Server) SearchStatVar(
 	if err != nil {
 		return nil, err
 	}
-	if len(localResp.StatVars) == 0 && len(localResp.Matches) == 0 {
-		if s.metadata.RemoteMixerDomain != "" {
-			remoteResp := &pb.SearchStatVarResponse{}
-			if err := util.FetchRemote(
-				s.metadata,
-				s.httpClient,
-				"/v1/variable/search",
-				in,
-				remoteResp,
-			); err != nil {
-				return nil, err
-			}
-			return remoteResp, nil
+
+	remoteResp := &pb.SearchStatVarResponse{}
+	if s.metadata.RemoteMixerDomain != "" {
+		if err := util.FetchRemote(
+			s.metadata,
+			s.httpClient,
+			"/v1/variable/search",
+			in,
+			remoteResp,
+		); err != nil {
+			return nil, err
 		}
 	}
-	return localResp, nil
+
+	resp := &pb.SearchStatVarResponse{
+		StatVars: append(localResp.StatVars[:], remoteResp.StatVars...),
+		Matches:  append(localResp.Matches[:], remoteResp.Matches...),
```

Comment:
The proto definition [mentions](https://github.com/gmechali/mixer/blob/8a5d4ee1a4cbe4b3a757aaf37040082f31a4e1e7/proto/stat_var.proto#L237) that matches should be unique strings. Can we ensure there are no duplicates in the merged result?

Reply:
Done.